gcp_pilot/storage.py

```diff
@@ -140,5 +141,28 @@ def _download(self, url: str) -> io.BytesIO:
     def get_uri(self, blob: Blob) -> str:
         return f"gs://{blob.bucket.name}/{blob.name}"
 
+    def generate_download_signed_url_v4(self, bucket_name: str, blob_name: str, minutes: int = 5):
+        """Generates a v4 signed URL for downloading a blob.
+
+        Doc: https://cloud.google.com/storage/docs/access-control/signing-urls-with-helpers#storage-signed-url-object-python
+
+        Note that this method requires a service account key file. You can not use
+        this if you are using Application Default Credentials from Google Compute
+        Engine or from the Google Cloud SDK.
+        """
+
+        bucket = self.client.bucket(bucket_name)
+        blob = bucket.blob(blob_name)
+
+        url = blob.generate_signed_url(
+            version="v4",
+            # This URL is valid for x minutes
+            expiration=datetime.timedelta(minutes=minutes),
+            # Allow GET requests using this URL.
+            method="GET",
+        )
+
+        return url
```

Comment:
Considering that gpc-pilot is a wrapper for GCP API, we should add methods that are as friendly (and simple) as possible, but highly customizable.

Since GCS library is already very good and friendly, we do not have too much to wrap around it.
So my suggestion is to simplify the method name and keep the parameters flexible.
The comments too do not make much sense for gcp-pilot because the [authentication is way more robust](https://github.com/flamingo-run/gcp-pilot#identification-features) than it is stated.

```suggestion
    def get_download_url(self, bucket_name: str, blob_name: str, expiration: timedelta = timedelta(minutes=minutes), version: str = "v4"):
        bucket = self.client.bucket(bucket_name)
        blob = bucket.blob(blob_name)
        return blob.generate_signed_url(
            version=version,
            expiration=expiration,
            method="GET",
        )
```